src/pages/iou/request/step/IOURequestStepParticipants.tsx

```diff
@@ -82,18 +84,28 @@ function IOURequestStepParticipants({
     const receiptPath = transaction?.receipt?.source;
     const receiptType = transaction?.receipt?.type;
     const isAndroidNative = getPlatform() === CONST.PLATFORM.ANDROID;
-    const isMobileSafari = isMobileSafariUtil();
+    const isMobileSafari = isMobileSafariBrowser();
 
     // When the component mounts, if there is a receipt, see if the image can be read from the disk. If not, redirect the user to the starting step of the flow.
     // This is because until the expense is saved, the receipt file is only stored in the browsers memory as a blob:// and if the browser is refreshed, then
     // the image ceases to exist. The best way for the user to recover from this is to start over from the start of the expense process.
     // skip this in case user is moving the transaction as the receipt path will be valid in that case
     useEffect(() => {
-        if (isMovingTransactionFromTrackExpense(action)) {
+        if (isMovingTransactionFromTrackExpense) {
             return;
         }
         navigateToStartStepIfScanFileCannotBeRead(receiptFilename ?? '', receiptPath ?? '', () => {}, iouRequestType, iouType, transactionID, reportID, receiptType ?? '');
-    }, [receiptType, receiptPath, receiptFilename, iouRequestType, iouType, transactionID, reportID, action]);
+    }, [receiptType, receiptPath, receiptFilename, iouRequestType, iouType, transactionID, reportID, isMovingTransactionFromTrackExpense]);
+
+    // When the step opens, reset the custom unit of the draft transaction if it's moving from Track Expense.
+    // This is needed to revert the rate back to the original FAKE_P2P_RATE_ID when changing the destination workspace.
```

Comment:
Thanks for explaining why this is needed. I think we should explain a bit more. As I understand it we need to reset the custom unit to use the P2P rate when we change the workspace because the code must first check if the p2p rate exists on the workspace, and if not show an error message to force the user to choose a valid rate from the workspace.